AI_VoiceOver/VoiceOver.lua

```diff
@@ -210,8 +228,38 @@ function Addon:GOSSIP_SHOW()
     elseif GetGossipOptions then
         lastGossipOptions = { GetGossipOptions() }
     end
+
+    -- Save current NPC for dialog-frame sync option
+    currentGossipNPC = {
+        guid = guid,
+        name = targetName,
+    }
+end
+
+function Addon:QUEST_FINISHED()
+    if Addon.db.profile.Audio.StopAudioOnDisengage and currentQuest then
+        for index, queuedSound in ipairs(self.soundQueue.sounds) do
+            if queuedSound.questID == currentQuest.questID then
+                self.soundQueue:RemoveSoundFromQueueUsingIndex(index)
```

Comment:
Why create a separate function for deleting via index, when you could just call `RemoveSoundFromQueue(queuedSound)`?

Reply:
The idea was that `QUEST_FINISHED` and `GOSSIP_CLOSED` will need to loop through the sound queue and find the correct corresponding sound index anyway. Passing the entire sound data would require two loops through the queue when only one is needed. 

Since the number of queued up sounds are small, it probably doesn't make much difference, however in practice it seemed like a good idea. `RemoveSoundFromQueue` can be used exactly as before.

Reply:
> Passing the entire sound data would require two loops through the queue

How so?

Reply:
I know it's outdated at this point, but the reason was due to needing to iterate through the queue to match the soundData with the saved guid/event pair. Once the correct soundData was found, the `RemoveSoundFromQueue` would iterate again to actually remove the sound from the queue. So my options were to either duplicate the removal logic within my iteration, or create a new function that removes based on the index found from the first iteration. I chose to latter as I could simply refactor the existing removal function to make it work cleanly for both use cases.

However, as you pointed out, had I simply saved the entire soundData (as is done currently), I could just pass that into the original `RemoveSoundFromQueue`. 

TLDR: I thought I was being space-efficient and clever by not copying an entire table, but I ended up making it more complicated than it needed to be!

Reply:
Gotcha.